sycl/doc/CompilerAndRuntimeDesign.md

```diff
@@ -434,6 +434,52 @@ unit)
 * `per_kernel` - enables emitting a separate module for each kernel
 * `off` - disables device code split
 
+#### CUDA support
+
+The driver supports compilation to CUDA when the `nvptx64-nvidia-cuda-sycldevice` is passed to `-fsycl-targets`.
+
+Unlike other AOT targets, intermediate objects never go through SPIR-V. Instead all separate compilation objects compile down to llvm bitcode. All produced bitcode depends on two libraries, `libdevice.bc` (provided by the CUDA SDK) and `libspirv-nvptx64--nvidiacl.bc` (built by the libclc project).
+
+During the device linking step (device linker box in the [Separate Compilation and Linking](#separate-compilation-and-linking) illustration), llvm bitcode objects for the CUDA target are linked together alongside `libspirv-nvptx64--nvidiacl.bc` and `libdevice.bc`, compiled to PTX using the NVPTX backend, and assembled into a cubin using the `ptxas` tool (part of the CUDA SDK). The PTX file and cubin are assembled together using `fatbinary` to produce a CUDA fatbin. The CUDA fatbin is then passed to the offload wrapper tool.
+
+##### NVPTX Builtins
+
+When using the NVPTX target for SYCL, the compiler will expose NVPTX builtins supported by clang.
+
```

Comment:
Please elaborate a bit what this means. This does not mean NVPTX builtins can be used in the app's CPP source or in generic SYCL device RT, right? Otherwise compilation for other targets would fail. Maybe an example of an NVPTX builtin and how it is used?

Reply:
I clarified and added an example

Reply:
thanks, now it is clear